packages/loader/container-loader/src/container.ts

```diff
@@ -1879,4 +1878,22 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     private logContainerError(warning: ContainerWarning) {
         this.mc.logger.sendErrorEvent({ eventName: "ContainerWarning" }, warning);
     }
+
+    /**
+     * Set the connected state of the ContainerContext
+     * This controls the "connected" state of the ContainerRuntime as well
+     * @param state - Is the container currently connected?
+     * @param readonly - Is the container in readonly mode?
+     */
+    private setContextConnectedState(state: boolean, readonly: boolean): void {
```

Comment:
Something to consider - This function can actually compute both `state` and `readonly`. Might be more robust to do so rather than having callers compute them and pass them in?  There's pros/cons both ways, up to you.